axelor-account/src/main/java/com/axelor/apps/account/service/AccountService.java

```diff
@@ -274,6 +278,23 @@ public Account fillAccountCodeOnImport(Account account, int lineNo) throws Axelo
     return account;
   }
 
+  public List<AnalyticAccount> getAnalyticAccounts(Account account) {
+    List<AnalyticAccount> analyticAccountList = new ArrayList<AnalyticAccount>();
+    List<Long> analyticRulesIdsList = getRulesIds(account);
+
+    if (CollectionUtils.isNotEmpty(analyticRulesIdsList)) {
+      for (Long analyticRuleId : analyticRulesIdsList) {
+        AnalyticRules analyticRule = analyticRulesRepository.find(analyticRuleId);
+        if (analyticRule.getAnalyticAccountSet() != null) {
+          for (AnalyticAccount analyticAccount : analyticRule.getAnalyticAccountSet()) {
+            analyticAccountList.add(analyticAccount);
+          }
+        }
+      }
+    }
```

Comment:
Isn't is possible to directly fetch those accounts instead of two for loops ? You could put the query part of `getRulesIds` in a method and use it in an `analyticAccount` query in a new method for example.
Also I think there can be duplicates in the list.